### PR TITLE
domain: avoit to print too many log if the ddl job of runaway table is not finished

### LINKED ISSUE #52048: many runaway errors in log
## Bug Report

Please answer these questions before submitting your issue. Thanks!

### 1. Minimal reproduce step (Required)

checked any log

### 3. What did you see instead (Required)

for every second

```
[2024/03/23 10:25:46.302 +08:00] [ERROR] [runaway.go:147] ["try to get new runaway watch"] [error="[schema:1146]Table 'mysql.tidb_runaway_watch' doesn't exist"]
```

### PR DESCRIPTION
**Pull request by CabinfeverB**
…nished

### What problem does this PR solve?

Issue Number: close #52048

Problem Summary:

### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [x] Manual test (add detailed scripts or steps below)
> 
before
> tiup playground nightly --tag upgrade3 --db 3
> ctrl + C
> tiup playground nightly --tag upgrade3 --db 3
image

after
> tiup playground v7.1.1 --tag upgrade --db 3
> ctrl + C
> tiup playground nightly --tag upgrade --db 3 --db.binpath=~/github/tidb/bin/tidb-server
image

- [ ] No need to test
  > - [ ] I checked and no code files have been changed.
  > 

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

File: pkg/domain/resourcegroup/runaway.go [hunk 1]
```diff
@@ -282,6 +290,13 @@ func (rm *RunawayManager) markQuarantine(resourceGroupName, convict string, watc
 	}
 	// Add record without ID into watch list in this TiDB right now.
 	rm.addWatchList(record, ttl, false)
+	if !rm.syncerInitialized.Load() {
+		logutil.BgLogger().Warn("runaway syncer is not initialized, so can't record runaway watch",
```
Comment (lance6716) [inline pkg/domain/resourcegroup/runaway.go:294]: will this log be too frequent as well?

File: pkg/domain/resourcegroup/runaway.go [hunk 2]
```diff
@@ -380,6 +395,12 @@ func (rm *RunawayManager) getWatchFromWatchList(key string) *QuarantineRecord {
 
 func (rm *RunawayManager) markRunaway(resourceGroupName, originalSQL, planDigest string, action string, matchType RunawayMatchType, now *time.Time) {
 	source := rm.serverID
+	if !rm.syncerInitialized.Load() {
+		logutil.BgLogger().Warn("runaway syncer is not initialized, so can't record runaway queries",
```
Comment (lance6716) [inline pkg/domain/resourcegroup/runaway.go:399]: ditto

File: pkg/domain/runaway.go [hunk 3]
```diff
@@ -140,20 +141,49 @@ func (do *Domain) deleteExpiredRows(tableName, colName string, expiredDuration t
...
+			err = do.updateNewAndDoneWatch()
+			if err == nil {
+				do.wg.Run(do.runawayRecordFlushLoop, "runawayRecordFlushLoop")
+				do.wg.Run(do.runawayWatchSyncLoop, "runawayWatchSyncLoop")
+				do.runawayManager.MarkSyncerInitialized()
+				return
+			}
+		}
+		count++
+		if count == 1800 {
```
Comment (lance6716) [inline pkg/domain/runaway.go:167]: every 1800 seconds?

**Review by lance6716 — COMMENTED** [review]
How do you think we choose another way to fix, which is starting the runaway functionality after the bootstrap

**Comment by CabinfeverB** [comment]
> How do you think we choose another way to fix, which is starting the runaway functionality after the bootstrap

Your idea is the best way, but I don't know much about the bootstrap process, so I'm not sure how to do. This is only a workaround.

File: pkg/domain/runaway.go (hunk 3 above)
Comment (CabinfeverB) [reply pkg/domain/runaway.go:167]: Yes, I think we should log it every appropriate time interval

**Review by lance6716 — APPROVED** [review]

**Review by glorv — APPROVED** [review]

**Pull request merged**